### PR TITLE
Rebase CircleCI to master if it is gcc5_4

### PR DESCRIPTION
**Pull request by houseroad**
This is the first step of adding CI for bc breaking changes detection of function shcemas.

### CODE REVIEW

File: .circleci/verbatim-sources/pytorch-job-specs.yml [hunk 1]
```diff
@@ -19,6 +19,27 @@ jobs:
           docker pull ${DOCKER_IMAGE} >/dev/null
           export id=$(docker run -t -d -w /var/lib/jenkins ${DOCKER_IMAGE})
 
+          # Rebase to master only if in xenial_py3_6_gcc5_4 case
```
Comment (bddppq) [inline .circleci/verbatim-sources/pytorch-job-specs.yml]: Better to put this chunk of code into a "step"?

File: .circleci/cimodel/data/pytorch_build_data.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
         ]),
         ("gcc", [
             ("4.8", [X("3.6")]),
-            ("5.4", [
+            ("5.4", [  # All this subtree rebases to master and then build
```
Comment (bddppq) [inline .circleci/cimodel/data/pytorch_build_data.py]: Better to add `compat` to the name of jobs that run the backward compatibility checks? (and then configure the generator script to add a step to merge with master for all `compat` jobs)

Comment (houseroad) [reply .circleci/cimodel/data/pytorch_build_data.py]: Yes. I will add bc to the name of test phase, see https://github.com/pytorch/pytorch/pull/26329

This is only for build. Want to share the build stage with some other tests.

File: .circleci/verbatim-sources/pytorch-job-specs.yml (hunk 1 above)
Comment (houseroad) [reply .circleci/verbatim-sources/pytorch-job-specs.yml]: Sure, if that is preferred, I can turn it into a step.

**Review by ezyang — APPROVED** [review]
okey dokey, although this code is a lot more complicated than I expected lol

**Pull request closed without merging**